### PR TITLE
use ruff as default linter

### PR DESCRIPTION
**Pull request by adehad**
# Summary of changes

1. Replaces flake8 and isort with ruff, a more performant variant

## Interesting to note
1. lots of default plugin support with pyproject compatible config
2. has autofixing for lots of the rules, this is the main motivator to consider this package, we want to spend less time addressing lint errors and more time actually writing code

### CODE REVIEW

File: jira/client.py [hunk 1]
```diff
@@ -2632,10 +2632,9 @@ def issue_type(self, id: str) -> IssueType:
         return self._find_for_resource(IssueType, id)
 
     def issue_type_by_name(self, name: str, project: Optional[str] = None) -> IssueType:
-        """
-        Args:
+        """Args:
```
Comment (adehad) [inline jira/client.py:2635]: needs a summary to prevent this bad autoformat

**Comment by studioj** [comment]
a why, would be useful

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,10 +1,5 @@
 ---
 repos:
-  - repo: https://github.com/psf/black
-    rev: 22.10.0
-    hooks:
-      - id: black
-        language_version: python3
   - repo: https://github.com/pre-commit/pre-commit-hooks
```
Comment (adehad) [inline .pre-commit-config.yaml:3]: ruff doesn't replace black

File: tox.ini [hunk 3]
```diff
@@ -66,7 +66,7 @@ allowlist_externals =
 description = Update dependency lock files
 # Force it to use oldest supported version of python or we would lose ability
 # to get pinning correctly.
-basepython = python3.8
+; basepython = python3.8
```
Comment (adehad) [inline tox.ini:69]:
```suggestion
basepython = python3.8
```

**Comment by adehad** [comment]
@studioj  have updated the summary a bit, I was really impressed by the autofix capabilities at work so I think it would be a great addition to add to reduce manually addressing many of the lint errors that can be avoided.

**Comment by studioj** [comment]
Sounds ok for me on the other hand I really like black 😂

**Comment by adehad** [comment]
> Sounds ok for me on the other hand I really like black 😂

yep that's a mistake on my end, this tool is meant to be used together with black for now

File: pyproject.toml [hunk 4]
```diff
@@ -38,3 +32,51 @@ python_version = 3.8
...
+select = [
+    "E", # pydocstyle
+    "W", # pydocstyle
+    "F", # pyflakes
+    "I", # isort
+    "UP", # pyupgrade
+    "D", # docstrings
+]
+ignore = [
+    "E501", # We have way to many "line too long" errors at the moment
```
Comment (adehad) [inline pyproject.toml:46]:
```suggestion
    "E501", # We have way too many "line too long" errors at the moment
```

File: tests/ruff.toml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+extend = "../pyproject.toml"
+ignore = [
+    "E501", # We have way to many "line too long" errors at the moment
```
Comment (adehad) [inline tests/ruff.toml:3]:
```suggestion
    "E501", # We have way too many "line too long" errors at the moment
```

File: pyproject.toml [hunk 6]
```diff
@@ -6,12 +6,6 @@ build-backend = "setuptools.build_meta"
 # Equivalent to use_scm_version=True
 [tool.setuptools_scm]
 
-# Tool Config
-[tool.isort]
-profile = "black"
-src_paths = ["jira", "tests"]
```
Comment (studioj) [inline pyproject.toml:12]: is this not needed anymore (for black .... I didnt study yet on the pyproject.toml config stuff)

Comment (adehad) [reply pyproject.toml:12]: not needed as `ruff` implements `isort` so we have a `tool.ruff.isort` section

**Review by studioj — APPROVED** [review]

**Comment by adehad** [comment]
@studioj feel free to merge (you should have the permissions!)

**Pull request merged**